pkg/azure/keyvault/store.go

```diff
@@ -56,16 +54,48 @@ func (s *Store) Connect() error {
 	return nil
 }
 
-func (s *Store) Resolve(keyName string, keyValue string) (string, error) {
+func (s *Store) Resolve(ctx context.Context, keyName string, keyValue string) (string, error) {
+	ctx, log := tracing.StartSpan(ctx)
+	defer log.EndSpan()
+	log.SetAttributes(attribute.String("secret name", keyValue))
+
 	if strings.ToLower(keyName) != SecretKeyName {
-		return s.hostStore.Resolve(keyName, keyValue)
+		return s.hostStore.Resolve(ctx, keyName, keyValue)
+	}
+
+	if err := s.Connect(ctx); err != nil {
+		return "", err
 	}
 
 	secretVersion := ""
-	result, err := s.client.GetSecret(context.Background(), s.vaultUrl, keyValue, secretVersion)
+	result, err := s.client.GetSecret(ctx, s.vaultUrl, keyValue, secretVersion)
 	if err != nil {
-		return "", errors.Wrapf(err, "could not get secret %s from %s", keyValue, s.vaultUrl)
+		return "", log.Error(fmt.Errorf("could not get secret %s: %w", keyValue, err))
 	}
 
 	return *result.Value, nil
 }
+
+// Create saves a the secret to azure's keyvault using the keyValue as the
+// secret key.
+// It implements the Create method on the secret plugins' interface.
+func (s *Store) Create(ctx context.Context, keyName string, keyValue string, value string) error {
+	ctx, log := tracing.StartSpan(ctx)
```

Comment:
Just like you are doing for resolve, let's add an attribute for the key that we are persisting